src/go/rpk/pkg/cli/cluster/config/edit_test.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEditorAndOptions(t *testing.T) {
+	testCases := map[string]struct {
+		editor   string
+		fileName string
+		want     []string
+	}{
+		"default": {
+			editor:   "code -w",
+			fileName: "/tmp/one.yaml",
+			want:     []string{"code", "-w", "/tmp/one.yaml"},
+		},
+		"noOptions": {
+			editor:   "code",
+			fileName: "/tmp/one.yaml",
+			want:     []string{"code", "/tmp/one.yaml"},
+		},
+		"withQuotedOption": {
+			editor:   "foo 'why would I do this'",
+			fileName: "/tmp/one.yaml",
+			want:     []string{"foo", "'why would I do this'", "/tmp/one.yaml"},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			got, err := editorAndArguments(tc.editor, tc.fileName)
```

Comment:
@rockwotj  - in the latest commit I tried using shellquote, I see the split works fine but then I see the quotes are stripped with the split.  I have added a test for edit and the `TestEditorAndOptions` will still fail with quotes.

Test fail output

```shell
--- FAIL: TestEditorAndOptions (0.00s)
    --- FAIL: TestEditorAndOptions/withQuotedOption (0.00s)
      .../git/redpanda-data/redpanda/src/go/rpk/pkg/cli/cluster/config/edit_test.go:38: want []string{"foo", "'why would I do this'", "/tmp/one.yaml"} but got []string{"foo", "why would I do this", "/tmp/one.yaml"}
```

Reply:
Oh sorry yes the quotes should be removed, so the library is doing the right thing. Otherwise this looks good - thanks for the patch!

Reply:
sorry I never used that lib so want to make sure. Thanks for the clarification.